### ISSUE TITLE
layers benchmarks are failing with "key not found"

### ISSUE THREAD

**Issue by samoht** [mirage/irmin#1164]
The easiest way to repro:

```
$ dune exec -- ./bench/irmin-pack/layers.exe -n 3000000
[... wait ~30s ...]
Benchmarks for layered store: internal error, uncaught exception:
                              (Failure "a65756b629e51a22451d93a5e4a05fef6685d5c0: unknown key")
```

**Comment by samoht** [comment]
I'm trying to bisect the issue without much success so far.

**Comment by samoht** [comment]
Fixed by reverting #1165
